independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -112,7 +112,8 @@ public <T> T get(Contextual<T> contextual) {
 
     @Override
     public boolean isActive() {
-        return currentContext.get() != null;
+        RequestContextState requestContextState = currentContext.get();
+        return requestContextState == null ? false : requestContextState.isValid();
```

Comment:
It's fine, the only downside is the dependent load, but having a volatile load is pretty check on most architecture, and regardless... there is not much to do here, given that we requires correctness first!